CMakeLists.txt

```diff
@@ -0,0 +1,487 @@
+#
+# Copyright (c) 2022, Arm Limited and affiliates.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+# How to use this file
+#
+# This file is used to build LLVM Embedded Toolchain for Arm.
+# Recent versions of the following tools are pre-requisites:
+# * A toolchain such as gcc & binutils
+# * cmake
+# * meson
+# * ninja
+# * python3
+# * make and qemu to run tests
+#
+# Commands to build:
+#   mkdir build
+#   cd build
+#   cmake .. -GNinja -DCMAKE_BUILD_TYPE=Release -DFETCHCONTENT_QUIET=OFF
+#   ninja
+#   ninja check-llvmembeddedtoolchainforarm
+#
+# To make it easy to get started, the above command checks out
+# llvm-project & picolibc Git repos automatically.
+#
+# If the repos are checked out automatically then cmake will fetch the
+# latest changes and check them out every time it runs. To disable this
+# behaviour run:
+#   cmake . -DFETCHCONTENT_FULLY_DISCONNECTED=ON
+#
+# If you prefer you can check out and patch the repos manually and use those:
+#   mkdir repos
+#   git -C repos clone https://github.com/llvm/llvm-project.git
+#   git -C repos/llvm-project apply ../../patches/llvm-HEAD.patch
+#   git -C repos clone https://github.com/picolibc/picolibc.git
+#   git -C repos/picolibc apply ../../patches/picolibc-HEAD.patch
+#   mkdir build
+#   cd build
+#   cmake .. -GNinja -DCMAKE_BUILD_TYPE=Release -DFETCHCONTENT_SOURCE_DIR_LLVMPROJECT=../repos/llvm-project -DFETCHCONTENT_SOURCE_DIR_PICOLIBC=../repos/picolibc
+#   ninja
+#   ninja check-llvmembeddedtoolchainforarm
+#
+# To install the toolchain run:
+#   cmake . --install-prefix /absolute/path/to/install/directory
+#   ninja install-llvmembeddedtoolchainforarm
+#
+# This file is designed to be used in a way that will be familiar to
+# LLVM developers. Targets like clang and check-all can be built as usual.
+# In addition there are targets to build picolibc & runtimes variants.
+
+
+# CONFIGURE_HANDLED_BY_BUILD was introduced in CMake 3.20 and it
+# greatly speeds up incremental builds.
+cmake_minimum_required(VERSION 3.20)
+
+project(
+    LLVMEmbeddedToolchainForArm
+    DESCRIPTION "LLVM Embedded Toolchain for Arm"
+    HOMEPAGE_URL "https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm"
+)
+
+set(BUG_REPORT_URL "https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm/issues" CACHE STRING "")
+set(LLVM_CCACHE_BUILD ON CACHE BOOL "")
```

Comment:
Either make this `OFF` by default or add `ccache` into the prerequisites.

Reply:
Good point. I've changed it so it only enables ccache by default if ccache can be found. And it can always be overridden by running cmake -DLLVM_CCACHE_BUILD=OFF